core/jobs/holds_notification.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from core.config import Configuration, ConfigurationConstants
+from core.model import Base
+from core.model.configuration import ConfigurationSetting
+from core.model.patron import Hold
+from core.monitor import SweepMonitor
+from core.util.notifications import PushNotifications
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Query
+
+    from core.model.collection import Collection
+
+
+class HoldsNotificationMonitor(SweepMonitor):
+    """Sweep across all holds that are ready to be checked out by the user (position=0)"""
+
+    MODEL_CLASS: type[Base] | None = Hold
+    SERVICE_NAME: str | None = "Holds Notification"
+
+    def run_once(self, *ignore):
+        setting = ConfigurationSetting.sitewide(
+            self._db, Configuration.PUSH_NOTIFICATIONS_STATUS
+        )
+        if setting.value == ConfigurationConstants.FALSE:
+            self.log.info(
+                "Push notifications have been turned off in the sitewide settings, skipping this job"
+            )
+            return
+        return super().run_once(*ignore)
```

Comment:
What happens if this is turned on, but the configuration file is not set or is not present?

Ideally, we'd want to log an error indicating the condition, but not have the jobs limp along throwing an error for each hold (in this case), loan, or patron sync notification.

Reply:
The README states the jobs will run with an 'unset' setting, which basically means True is the default. I can switch it around to be False as default if this means the implementation teams will have their hands full during the deployment